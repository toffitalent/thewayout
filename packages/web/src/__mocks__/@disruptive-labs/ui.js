const actual = jest.requireActual('@disruptive-labs/ui');

const showToast = jest.fn();

actual.Toast.show = showToast;
actual.Toast.danger = showToast;
actual.Toast.info = showToast;
actual.Toast.success = showToast;
actual.Toast.warning = showToast;

module.exports = {
  ...actual,
  showToast,
};
