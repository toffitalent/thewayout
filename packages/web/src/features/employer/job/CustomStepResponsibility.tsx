import {
  Button,
  Field,
  Flex,
  Text,
  useFieldArray,
  useForm,
  useWizardContext,
} from '@disruptive-labs/ui';
import TrashIcon from '@disruptive-labs/ui/dist/icons/Trash';
import { forwardRef } from 'react';
import { WizardData } from './createJobDataTypes';

export const CustomStepResponsibilities = forwardRef((_, ref: any) => {
  const { next, data } = useWizardContext<WizardData>();

  const { submitForm, control, register, watch } = useForm({
    defaultValues: {
      responsibilities: !data.responsibilities?.length
        ? [{ value: '' }]
        : data.responsibilities.map((el) => ({ value: el })),
    },
    onSubmit: async ({ responsibilities }: { responsibilities: { value: string }[] }) => {
      await next({
        ...data,
        responsibilities: responsibilities.map((el) => el.value).filter((resp) => !!resp),
      });
    },
  });

  const { fields, append, remove } = useFieldArray({ control, name: 'responsibilities' });
  const watchFieldArray = watch('responsibilities');
  const controlledFields = fields.map((field, index) => ({
    ...field,
    ...watchFieldArray[index],
  }));

  return (
    <form onSubmit={submitForm}>
      <Flex textAlign="left" maxWidth={96} ml="auto" mr="auto">
        {controlledFields.map((responsibility, index) => (
          <Flex mb={3} key={responsibility.id}>
            <Flex mb={3} direction="row" justify="space-between">
              <Text fontWeight="700">{`Responsibility #${index + 1}`}</Text>
              {responsibility && responsibility.value !== '' && (
                <TrashIcon
                  height={6}
                  width={6}
                  color="primary"
                  onClick={() => {
                    remove(index);
                    if (controlledFields.length === 1) append({ value: '' });
                  }}
                />
              )}
            </Flex>
            <Field
              id={responsibility.id}
              {...register(`responsibilities.${index}.value` as const)}
              type="text"
              multiline
              placeholder="E.g. Always be ready to assist customers with menu options and payment."
              fluid
              control={control}
              ref={ref}
              helperText="Max 1,000 characters."
              validate={{ maxLength: 1000 }}
            />
          </Flex>
        ))}
        <Button
          variant="outline"
          onClick={() => append({ value: '' })}
          mt={5}
          colorScheme="primary"
          disabled={!controlledFields[controlledFields.length - 1]?.value.length}
        >
          Add more
        </Button>
      </Flex>
      <Button colorScheme="primary" fluid type="submit" maxWidth={96} mt={8}>
        Save & Next
      </Button>
    </form>
  );
});
