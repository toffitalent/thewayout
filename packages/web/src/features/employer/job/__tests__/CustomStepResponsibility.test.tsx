import { useWizardContext } from '@disruptive-labs/ui';
import { render, waitFor } from '@test';
import { CustomStepResponsibilities } from '../CustomStepResponsibility';

jest.mock('@disruptive-labs/ui', () => ({
  ...jest.requireActual('@disruptive-labs/ui'),
  useWizardContext: jest.fn(),
}));

describe('CustomStepResponsibilities', () => {
  beforeEach(() => {
    (useWizardContext as any).mockImplementation(() => ({
      data: {
        responsibilities: [],
      },
      setData: jest.fn(),
      goTo: jest.fn(),
    }));
  });

  test('renders correctly', async () => {
    const { container, findByPlaceholderText } = render(<CustomStepResponsibilities />);
    await waitFor(async () =>
      expect(
        await findByPlaceholderText(
          'E.g. Always be ready to assist customers with menu options and payment.',
        ),
      ).toBeInTheDocument(),
    );
    expect(container).toMatchSnapshot();
  });
});
