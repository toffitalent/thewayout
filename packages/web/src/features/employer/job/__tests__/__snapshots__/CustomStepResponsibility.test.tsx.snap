// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomStepResponsibilities renders correctly 1`] = `
<div>
  <form>
    <div
      class="maxWidth-96 marginRight-auto marginLeft-auto textAlign-left"
    >
      <div
        class="marginBottom-3"
      >
        <div
          class="flexDirection-row justifyContent-space-between marginBottom-3"
        >
          <p
            class="text fontWeight-700"
          >
            Responsibility #1
          </p>
        </div>
        <div
          class="control"
          role="group"
        >
          <div
            class="root primary md fluid md multiline"
          >
            <textarea
              aria-describedby=":r0:-helptext"
              class="input"
              id=":r0:"
              name="responsibilities.0.value"
              placeholder="E.g. Always be ready to assist customers with menu options and payment."
              rows="4"
              type="text"
            />
          </div>
          <div
            aria-live="polite"
            class="helperText"
            id=":r0:-helptext"
          >
            Max 1,000 characters.
          </div>
        </div>
      </div>
      <button
        class="button md outline c-primary disabled marginTop-5"
        disabled=""
        type="button"
      >
        Add more
      </button>
    </div>
    <button
      class="button md solid c-primary fluid maxWidth-96 marginTop-8"
      type="submit"
    >
      Save & Next
    </button>
  </form>
</div>
`;
